gene_gene/gene_interacts_gene/dataset_statistics/dataset_statistics.py

```diff
@@ -0,0 +1,520 @@
+        total_candidates_df[["gene1_id", "gene2_id"]].drop_duplicates().shape[0]
+    )
+)
+print(
+    "Total number of unique candidates after filter: {}".format(
+        total_candidates_df.query("sen_length < 83+1")[["gene1_id", "gene2_id"]].drop_duplicates().shape[0]
+    )
+)
+print("Total number of unique candidates being thrown out: {}".format(len(before_filter.difference(after_filter))))
+
+
+# In[23]:
+
+
+filtered_total_candidates_df = total_candidates_df.query("sen_length < 83+1")
+
+
+# In[24]:
+
+
+venn2(
+    [
+        set(filtered_total_candidates_df.query("hetionet==0").sentence_id),
+        set(filtered_total_candidates_df.query("hetionet==1").sentence_id)
+    ], set_labels=["Not In Hetionet", "In Hetionet"])
+plt.title("# of Unique Sentences in Entire Dataset with Co-Mention Pair in/not in hetionet")
+
+
+# # Co-occuring Mentions Sentence Stats
+
+# This next block contains a visualization about the number of mentions a sentence may contain.
+
+# In[25]:
+
+
+sns.distplot(
+    filtered_total_candidates_df
+    .drop_duplicates("sentence_id")["gene_mention_count"], 
+    kde=False, 
+    rug=False,
+    bins=20
+)
+
+
+# In[26]:
+
+
+filtered_total_candidates_df.drop_duplicates("sentence_id")["gene_mention_count"].value_counts()
+
+
+# The above graph shows the distribution of gene mentions per sentence. Looking above majority of the sentences contain only two unique gene mentions, while a select number of sentences can contain a total of 25 gene mentions. Overall, this is an acceptable distribution, since majority of sentences will have just two mention pairs.
+
+# # Subsampled Set Distributions
+
+# Take a look at the stratification for each train/dev/test set.
+
+# ## Training Set
+
+# Here the training set covers majority of the characterisitics are the entire dataset. This isn't surprising because a stratified sort was used to allocate candidates.
+
+# In[27]:
+
+
+sns.distplot(
+    filtered_total_candidates_df
+    .query("split==3")
+    .drop_duplicates("sentence_id")["gene_mention_count"], 
+    kde=False, 
+    rug=False,
+    bins=20
+)
+
+
+# In[28]:
+
+
+filtered_total_candidates_df.query("split==3").drop_duplicates("sentence_id")["gene_mention_count"].value_counts()
+
+
+# In[29]:
+
+
+sns.distplot(filtered_total_candidates_df.query("split==3")["sen_length"], rug=False)
+
+
+# ## Dev Set
+
+# This block contains information on the development set (aka tuning set).
+
+# In[30]:
+
+
+sns.distplot(
+    filtered_total_candidates_df
+    .query("split==4")
+    .drop_duplicates("sentence_id")["gene_mention_count"], 
+    kde=False, 
+    rug=False,
+    bins=20
+)
+
+
+# In[31]:
+
+
+filtered_total_candidates_df.query("split==4").drop_duplicates("sentence_id")["gene_mention_count"].value_counts()
+
+
+# In[32]:
+
+
+sns.distplot(filtered_total_candidates_df.query("split==4")["sen_length"], rug=False)
+
+
+# ## Test Set
+
+# This block contains information on the test set.
+
+# In[33]:
+
+
+sns.distplot(
```

Comment:
Why is the test set shifted?

Reply:
As discussed the distribution only contains gene mentions that are greater than two. Hence the shifted appearance in this distribution.